### ISSUE TITLE
-d option broken?

### ISSUE THREAD

**Issue by nbecker** [amperser/proselint#624]
proselint -d outroute_timing_design.txt 
Deleting the cache...
rm: invalid option -- 'p'
Try 'rm --help' for more information.

 proselint -v
0.7.0

running on python-3.5.1

**Comment by m-charlton** [comment]
It looks as if the problem is in the `clear_cache` function in 
`proselint/command_line.py`:

```python
45   def clear_cache():
46    """Delete the contents of the cache."""
47    click.echo("Deleting the cache...")
48    subprocess.call(["find", ".", "-name", "'*.pyc'", "-delete"])
49    subprocess.call([
50        "rm",
51        "-rfv",
52        "proselint/cache",
53        "&&",
54        "mkdir",
55        "-p",
56        os.path.join(os.path.expanduser("~"), ".proselint"),
57    ])
```

The `rm` command is executed in the subprocess call OK. For some reason the 
`-p` argument is being evaluated as part of the call to `rm` not `mkdir`.

One solution would be to divide the `rm` and `mkdir` operations into separate 
subprocess calls

```python
    result = subprocess.call(["rm", "-rfv", "proselint/cache"])

    if result == 0:
        subprocess.call(["mkdir", "-p", os.path.join(os.path.expanduser("~"), ".proselint")])
```

An alternative approach would be to use the existing `shutil.rmtree(...)` and
`os.mkdir(...)` calls:

```python

shutil.rmtree("proselint/cache", ignore_errors=False)

try:
    os.makedirs(os.path.join(os.path.expanduser("~"), ".proselint"))
except OSError as os_excp:
    # The exception can be thrown if `~/.proselint` already exists, in this 
    # case would we want to clean all caches bellow `~/.proselint`? Another
    # reason could be for access issues, though you'd hope the user would have 
    # access to their own home directory. The `pass` may not be a good idea here:
    pass
    
```

Questions:

- `shutil.rmtree` does not work with symbolic links to directories. Would this be a problem?
- If the `proselint/cache` is *not* removed then should the `~/.proselint` directory allowed to be created?
- See the comments in `except OSError` above. 

I'm quite prepared to fix this and add unit tests.

**Comment by Nytelife26** [comment]
Fixed in #655.
